docs/source/howto/codes/interfacing.rst

```diff
@@ -0,0 +1,282 @@
+.. _how-to:codes:plugin:
+
+Interfacing external codes
+==========================
+
+In order to run an external code, you will first have to explain to AiiDA how to run it: how to create inputs files from the content of the input node, how the program can read those input files, where to find and how to read the output files, etc.
+The way you transmit these instructions to AiiDA is by extending the |CalcJob| and the |Parser| classes:
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+    from aiida.engine import CalcJob
+
+    class ArithmeticAddCalculation(CalcJob):
+
+        @classmethod
+        def define(cls, spec):
+            super().define(spec)
+            # Specifications of inputs, outputs and exit codes.
+            # no return statement
+
+        def prepare_for_submission(self, folder):
+            # Instructions on how to prepare the input files
+            return calcinfo
+
```

Comment:
You should place all code segments into context. Just a brief sentence before the segment about what you are showing.

Reply:
Not super sure what you mean since I think this code is preceeded by an explanation of what is going to contain, but I modified it a bit with the hope that it is clearer now.